apps/cowswap-frontend/src/modules/orderProgressBar/updaters/OrderProgressStateUpdater.tsx

```diff
@@ -26,9 +32,20 @@ export function OrderProgressStateUpdater(): ReactNode {
   )
 
   useEffect(() => {
-    const trackedIds = account && chainId ? marketOrders.map((order) => order.id) : []
-    pruneProgressState(trackedIds)
-  }, [account, chainId, marketOrders, pruneProgressState])
+    const trackedIdsSet = new Set<string>()
+
+    if (account && chainId) {
```

Comment:
question: if there is no account/chainId - does it really make sense to handle any order progress state?

Reply:
> Yes, it still matters. Both the Surplus modal and the confirmation modal keep referencing order IDs even if the wallet briefly disconnects or switches chains, so we continue pruning/retaining their IDs to avoid dropping solver info during those transitions (see the comment added
  around OrderProgressStateUpdater.tsx lines 34‑50).